src/tacticsuggestions.ts

```diff
@@ -0,0 +1,38 @@
+import { commands, Disposable,  window } from 'vscode';
+import { Server } from './server';
+
+/** Pastes suggestions provided by tactics such as `squeeze_simp` */
+export class TacticSuggestions implements Disposable {
+    private subscriptions: Disposable[] = [];
+    constructor(private server: Server) {
+        const command = 'lean.pasteTacticSuggestion';
+
+        const commandHandler = (name: string = 'world') => {
+            const editor = window.activeTextEditor;
+
+            if (editor) {
+                const curFileName = editor.document.fileName;
+                const curPosition = editor.selection.active;
+                const curSelection = editor.selection;
+                const messages = this.server.messages
+                    .filter((m) => m.file_name === curFileName &&
+                        m.pos_line === curPosition.line + 1)
+                    .sort((a, b) => a.pos_col - b.pos_col);
+                // console.log(messages)
+                if (messages.length > 0) {
+                    editor.edit(editBuilder => {
+                        editBuilder.replace(curSelection, messages[0].text);
```

Comment:
```suggestion
                        editBuilder.replace(???, messages[0].text);
```

The range to be replaced should be calculated from the message.  Maybe `message.pos_col` until the next comma or the end of the line, whichever comes first?

Reply:
Just looking for the first comma won't work for `squeeze_simp [lemma1, lemma2]` or `tidy? { tactics := [tactic1, tactic2] }`. The "end of the line" idea could also fail for a long list of simp lemmas broken up with linebreaks.

Reply:
Maybe the `try this:` magic string should also include who sent the message. Something like `tidy says, try this:` and `simp says, try this`, etc... so that the replacement script has at least a little clue to look for what it should replace.

Reply:
Oooh, don't Lean tactics know which position in the file they are in? So `squeeze_simp` and `suggest` and friends could output coordinates together with the `try this:` message

Reply:
I implemented till the end of line for now.